include/nil/crypto3/zk/snark/systems/plonk/placeholder/permutation_argument.hpp

```diff
@@ -66,6 +68,20 @@ namespace nil {
                         typename permutation_commitment_scheme_type::precommitment_type permutation_poly_precommitment;
                     };
 
+                    static inline math::polynomial_dfs<typename FieldType::value_type> polynomial_product(
+                        std::vector<math::polynomial_dfs<typename FieldType::value_type>> multipliers)
+                    {
+                        while (multipliers.size() != 1) {

```

Comment:
Multiplication of dfs polynomials is faster if multiplied in a balanced way, I.E. muptiply each pair, then pairs of pairs etc.